lib/event_bus/services/observation.ex

```diff
@@ -44,19 +44,25 @@ defmodule EventBus.Service.Observation do
   @doc false
   @spec mark_as_completed(subscriber_with_event_ref()) :: :ok
   def mark_as_completed({subscriber, event_shadow}) do
-    {subscribers, completers, skippers} = fetch(event_shadow)
-    save_or_delete(event_shadow, {subscribers, [subscriber | completers], skippers})
+    case fetch(event_shadow) do
+      {subscribers, completers, skippers} ->
+        save_or_delete(event_shadow, {subscribers, [subscriber | completers], skippers})
+        nil -> :ok
+    end
   end
 
   @doc false
   @spec mark_as_skipped(subscriber_with_event_ref()) :: :ok
   def mark_as_skipped({subscriber, event_shadow}) do
-    {subscribers, completers, skippers} = fetch(event_shadow)
-    save_or_delete(event_shadow, {subscribers, completers, [subscriber | skippers]})
+    case fetch(event_shadow) do
+      {subscribers, completers, skippers} ->
+        save_or_delete(event_shadow, {subscribers, completers, [subscriber | skippers]})
+      nil -> :ok
```

Comment:
In here, we are loosing the context of why we are getting `nil` from the `ETS` storage. (the same as above)